### PR TITLE
feat(bigtable): Add MarshalJSON to allow clients to get a stringified version of the protobuf

### PR DESCRIPTION
**Pull request by ron-gal**

### CODE REVIEW

File: bigtable/type.go [hunk 1]
```diff
@@ -16,12 +16,26 @@ limitations under the License.
...
+	btapb "cloud.google.com/go/bigtable/admin/apiv2/adminpb"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+)
 
 // Type wraps the protobuf representation of a type. See the protobuf definition
 // for more details on types.
 type Type interface {
 	proto() *btapb.Type
+	ToJSON() (string, error)
```
Comment (codyoss) [inline bigtable/type.go:29]: Why is this needed in this library? This seems like code that could be written where it is needed as it does not access any private data to the interface.

Comment (ron-gal) [reply bigtable/type.go:29]: It's accessing proto() that is private.
We opted to go with https://medium.com/@chaewonkong/go-and-json-a-comprehensive-guide-to-working-with-json-in-golang-143fa2dfa897#:~:text=of%20an%20integer.-,Custom%20types,-%3A instead though :)

File: bigtable/type.go [hunk 2]
```diff
@@ -32,6 +36,14 @@ func (u unknown[T]) proto() *T {
 	return u.wrapped
 }
 
+// MarshalJSON returns the string representation of the protobuf.
+func (u unknown[T]) MarshalJSON() ([]byte, error) {
+	if t, ok := any(u.wrapped).(proto.Message); ok {
+		return protojson.MarshalOptions{}.Marshal(t)
```
Comment (codyoss) [inline bigtable/type.go:42]: Consider using the same marshal settings we use in GAPICs. Example: https://github.com/googleapis/google-cloud-go/blob/60ad7f32d69ed3986fb1326841bba2c5dc36e508/secretmanager/apiv1/secret_manager_client.go#L950

Comment (ron-gal) [reply bigtable/type.go]: Done

File: bigtable/type.go [hunk 3]
```diff
@@ -32,6 +39,14 @@ func (u unknown[T]) proto() *T {
 	return u.wrapped
 }
 
+// MarshalJSON returns the string representation of the protobuf.
+func (u unknown[T]) MarshalJSON() ([]byte, error) {
+	if t, ok := any(u.wrapped).(proto.Message); ok {
```
Comment (steveniemitz) [inline bigtable/type.go:44]: can we force wrapped to be a `proto.Message*` instead?

File: bigtable/type_test.go [hunk 4]
```diff
@@ -55,38 +109,29 @@ func TestStringProto(t *testing.T) {
...
+	gotJSON, err := MarshalJSON(st)
+	if err != nil {
+		t.Fatalf("Error calling ToJSON: %v", err)
 	}
-
-	got := AggregateType{Input: Int64Type{}, Aggregator: SumAggregator{}}.proto()
-	if !proto.Equal(got, want) {
-		t.Errorf("got type %v, want: %v", got, want)
+	wantJSON := "{\"stringType\":{\"encoding\":{\"utf8Raw\":{}}}}"
+	if strings.ReplaceAll(string(gotJSON), " ", "") != wantJSON {
```
Comment (steveniemitz) [inline bigtable/type_test.go:123]: rather than testing that the json equals a known string, I think we can instead just test that round tripping the object from json and back yield the "correct" thing?  Comparing json strings has the potential for false errors.

Comment (ron-gal) [reply bigtable/type_test.go]: Done

File: bigtable/type.go [hunk 5]
```diff
@@ -16,14 +16,40 @@ limitations under the License.
...
+)
 
 // Type wraps the protobuf representation of a type. See the protobuf definition
 // for more details on types.
 type Type interface {
 	proto() *btapb.Type
 }
 
+var marshalOptions = protojson.MarshalOptions{AllowPartial: true, UseEnumNumbers: true}
+var unmarshalOptions = protojson.UnmarshalOptions{AllowPartial: true, DiscardUnknown: true}
```
Comment (steveniemitz) [inline bigtable/type.go:32]: I dont think we want DiscardUnknown? assuming that does what it sounds like.

File: bigtable/type_test.go [hunk 6]
```diff
@@ -37,12 +39,54 @@ func aggregateProto() *btapb.Type {
...
+	it := Int64Type{Encoding: BigEndianBytesEncoding{}}
+	got := it.proto()
 	if !proto.Equal(got, want) {
 		t.Errorf("got type %v, want: %v", got, want)
 	}
+
+	gotJSON, err := MarshalJSON(it)
+	if err != nil {
+		t.Fatalf("Error calling MarshalJSON: %v", err)
+	}
```
Comment (steveniemitz) [inline bigtable/type_test.go:79]: I think this could be pulled out into a helper method? it's basically the same for each type right?

File: bigtable/type.go (hunk 5 above)
Comment (ron-gal) [reply bigtable/type.go]: Done
